File: 1063-best-sightseeing-pair/1063-best-sightseeing-pair.py
class Solution:
    def maxScoreSightseeingPair(self, values: List[int]) -> int:
        score=0
        max_i=values[0]
        for i in range(1,len(values)):
            # print(max_i)
            score=max(score,max_i+values[i]-i)
            max_i=max(max_i,values[i]+i)
        return score

File: 1063-best-sightseeing-pair/1063-best-sightseeing-pair.java
import java.util.*;
class Solution {
    public int maxScoreSightseeingPair(int[] values) {
        if(values.length == 0)
        {
            return 0;
        }
        int maxVal = values[0];
        int maxScore = 0;
        for(int i = 1; i < values.length; i++)
        {
            maxVal = Math.max(maxVal,values[i-1] + i - 1);
            maxScore = Math.max(maxScore,maxVal + values[i] - i);
        }
        return maxScore;
    }
}